### PR TITLE
fix(cautils): use TrimPrefix to strip URL scheme in CreatePortForwarder

### LINKED ISSUE #2017: strings.TrimLeft in CreatePortForwarder strips valid hostname characters
## Description

`CreatePortForwarder` in `core/cautils/portforwarder.go` uses `strings.TrimLeft` to strip the URL scheme from the Kubernetes API server host:

```go
hostIP := strings.TrimLeft(k8sClient.K8SConfig.Host, "htps:/")
```

`strings.TrimLeft` treats its second argument as a **character set**, not a prefix. After it strips `https://`, it continues to strip *any* leading character that appears in the set `h`, `t`, `p`, `s`, `:`, `/`. This corrupts hostnames whose first character matches that set.

## Reproduction

| `K8SConfig.Host`                               | Current `hostIP`                  | Expected                          |
| ---------------------------------------------- | --------------------------------- | --------------------------------- |
| `https://1.2.3.4:6443`                         | `1.2.3.4:6443`                    | `1.2.3.4:6443` :white_check_mark: |
| `https://kubernetes.docker.internal:6443`      | `kubernetes.docker.internal:6443` | `kubernetes.docker.internal:6443` :white_check_mark: |
| `https://hello-cluster.example.com:6443`       | `ello-cluster.example.com:6443`   | `hello-cluster.example.com:6443` :x: |
| `https://test.example.com`                     | `est.example.com`                 | `test.example.com` :x:            |
| `https://prod.example.com`                     | `rod.example.com`                 | `prod.example.com` :x:            |
| `https://staging.example.com`                  | `taging.example.com`              | `staging.example.com` :x:         |

The corrupted host is then used as `serverURL.Host` for the SPDY dialer, so the port-forward to the kubescape operator pod will hit the wrong (or nonexistent) DNS name.

A minimal Go reproduction:

```go
package main

import (
	"fmt"
	"strings"
)

func main() {
	host := "https://hello-cluster.example.com:6443"
	fmt.Printf("%q\n", strings.TrimLeft(host, "htps:/"))
	// Output: "ello-cluster.example.com:6443"
}
```

## Impact

Cluster scans that rely on the in-cluster operator (`kubescape scan` against a remote cluster whose API server hostname starts with `h`, `t`, `p`, or `s`) silently dial the wrong host and the port-forward fails. Many real-world cluster names start with these letters — `prod-*`, `stg-*`, `test-*`, `hub-*` — so this is not a corner case.

## Fix

Replace `strings.TrimLeft` with `strings.TrimPrefix` for both `https://` and `http://`. I have a fix ready and will open a PR shortly.

### PR DESCRIPTION
**Pull request by sahitya-chandra**
## Overview

`CreatePortForwarder` in `core/cautils/portforwarder.go` used `strings.TrimLeft(host, "htps:/")` to strip the URL scheme from `K8SConfig.Host`. `TrimLeft` treats its second argument as a character set, so it also strips any leading `h`/`t`/`p`/`s`/`:`/`/` runes from the hostname itself — `https://hello.example.com:6443` became `ello.example.com:6443`, which then becomes the SPDY dialer target for the operator port-forward.

This PR introduces a small `stripScheme` helper that uses `strings.TrimPrefix` for `https://` and `http://`, and adds table-driven tests covering the regression cases (hostnames starting with `h`, `t`, `p`, `s`, plus IPs and `kubernetes.docker.internal`).

## How to Test

```
go test ./core/cautils/ -run Test_stripScheme -v
```

A reproducer for the original bug (no kubescape required):

```go
package main

import (
	"fmt"
	"strings"
)

func main() {
	host := "https://hello-cluster.example.com:6443"
	fmt.Printf("%q\n", strings.TrimLeft(host, "htps:/"))
	// "ello-cluster.example.com:6443"
}
```

## Related issues/PRs:

* Resolved #2017

## Checklist before requesting a review

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have added thorough tests.
- [x] New and existing unit tests pass locally with my changes

## Summary by CodeRabbit

* **Bug Fixes**
  * Port forwarding now correctly normalizes Kubernetes API server addresses, preserving any existing base path and handling URL schemes so connections use the correct server path.

* **Tests**
  * Added unit tests covering scheme and base-path handling in host normalization.

### CODE REVIEW

**Review by matthyx — COMMENTED** [review]
Left an inline suggestion for the remaining path-prefix case.

File: core/cautils/portforwarder.go [hunk 1]
```diff
@@ -35,9 +35,15 @@ func getPortForwardingPort() string {
...
+	host = strings.TrimPrefix(host, "https://")
+	host = strings.TrimPrefix(host, "http://")
+	return host
+}
+
 func CreatePortForwarder(k8sClient *k8sinterface.KubernetesApi, pod *v1.Pod, forwardingPort, namespace string) (OperatorConnector, error) {
 	path := fmt.Sprintf("/api/v1/namespaces/%s/pods/%s/portforward", namespace, pod.Name)
-	hostIP := strings.TrimLeft(k8sClient.K8SConfig.Host, "htps:/")
+	hostIP := stripScheme(k8sClient.K8SConfig.Host)
 	serverURL := &url.URL{Scheme: "https", Path: path, Host: hostIP}
```
Comment (matthyx) [inline core/cautils/portforwarder.go]: `rest.Config.Host` can also be a full base URL (for example `https://proxy.example.com/k8s`). In that case `stripScheme()` turns it into `proxy.example.com/k8s`, which isn't a valid `url.URL.Host` and also drops the `/k8s` prefix from the request path. Parsing only the scheme-bearing form keeps the existing `host:port` behavior intact while preserving any base path.

```suggestion
func splitHostAndBasePath(host string) (string, string, error) {
	if !strings.Contains(host, "://") {
		return host, "", nil
	}

	baseURL, err := url.Parse(host)
	if err != nil {
		return "", "", err
	}

	return baseURL.Host, strings.TrimRight(baseURL.Path, "/"), nil
}

func CreatePortForwarder(k8sClient *k8sinterface.KubernetesApi, pod *v1.Pod, forwardingPort, namespace string) (OperatorConnector, error) {
	path := fmt.Sprintf("/api/v1/namespaces/%s/pods/%s/portforward", namespace, pod.Name)
	hostIP, basePath, err := splitHostAndBasePath(k8sClient.K8SConfig.Host)
	if err != nil {
		return nil, err
	}
	serverURL := &url.URL{Scheme: "https", Path: basePath + path, Host: hostIP}
```

Could you also add a test for a prefixed host like `https://proxy.example.com/k8s`?

Comment (sahitya-chandra) [reply core/cautils/portforwarder.go]: Good catch sir, I hadn't considered the base-URL case. Applied your suggestion verbatim and added tests for `https://proxy.example.com/k8s`, the `host:port + path` variant, a trailing-slash case, and a multi-segment path. Pushing the fix now

**Review by matthyx — APPROVED** [review]
Rechecked after the follow-up commit. This addresses the path-prefix case, and the added coverage looks good.
